app/models/student.rb

```diff
@@ -2,21 +2,41 @@
 #
 # Table name: students
 #
-#  id                  :integer          not null, primary key
-#  created_at          :datetime
-#  updated_at          :datetime
-#  email               :string
-#  confirmation_token  :string
-#  confirmed           :boolean          default(FALSE)
-#  classroom_id        :integer
-#  survey_administered :boolean          default(FALSE)
+#  id                            :integer          not null, primary key
+#  created_at                    :datetime
+#  updated_at                    :datetime
+#  email                         :string
+#  confirmation_token            :string
+#  confirmed                     :boolean          default(FALSE)
+#  classroom_id                  :integer
+#  survey_administered           :boolean          default(FALSE)
+#  fundamentals_quiz_completed   :float            default(-1.0)
+#  water_quiz_completed          :float            default(-1.0)
+#  waste_quiz_completed          :float            default(-1.0)
+#  transportation_quiz_completed :float            default(-1.0)
+#  energy_quiz_completed         :float            default(-1.0)
+#  building_quiz_completed       :float            default(-1.0)
+#  health_quiz_completed         :float            default(-1.0)
+#  community_quiz_completed      :float            default(-1.0)
+#  application_quiz_completed    :float            default(-1.0)
+#  financial_quiz_completed      :float            default(-1.0)
 #
 
 class Student < ActiveRecord::Base
   belongs_to :classroom
 
   validates :email, presence: true, email: true
   validates :classroom_id, presence: true
+  validates :fundamentals_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :water_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :waste_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :transportation_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :energy_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :building_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :health_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :community_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :application_quiz_completed, numericality: { less_than_or_equal_to: 100 }
+  validates :financial_quiz_completed, numericality: { less_than_or_equal_to: 100 }
```

Comment:
These capture some of the constraints but not all of them. For example, this still allows quiz scores of -5.

We want to write a validation that only allows -1 and values in between 0-100.